src/Symfony/Component/Config/Definition/ArrayNode.php

```diff
@@ -209,8 +209,8 @@ public function getDefaultValue()
      */
     public function addChild(NodeInterface $node)
     {
-        $name = $node->getName();
-        if (!strlen($name)) {
+        $name = (string) $node->getName();
```

Comment:
\Symfony\Component\Config\Definition\BaseNode::__construct doesn't validate a type.

Addint validation will break some of UTs, I didn't look into details as very constraint in time.